flake.nix

```diff
@@ -1,25 +1,31 @@
 {
-
-  inputs.flake-utils.url = "github:numtide/flake-utils";
-
   outputs =
-    { flake-utils, self }:
-    flake-utils.lib.eachDefaultSystem (
-      system:
-      let
-        result = import ./default.nix { inherit system; };
-      in
-      {
-        packages = result.packages // {
-          default = result;
-        };
+    { self, nixpkgs }:
+    let
+      forAllSystems = nixpkgs.lib.genAttrs nixpkgs.lib.systems.flakeExposed;
 
-        apps.default = {
+      result = forAllSystems (system: import ./default.nix { inherit system; });
+    in
+    {
+      packages = forAllSystems (
+        system:
+        let
+          packages = result.${system}.packages;
+        in
+        packages // { default = packages.nixfmt; }
+      );
+
+      apps = forAllSystems (system: {
+        default = {
           type = "app";
-          program = "${result}/bin/nixfmt";
+          program = nixpkgs.lib.getExe self.packages.${system}.default;
         };
+      });
+
+      checks = forAllSystems (system: result.${system}.checks);
 
-        checks = result.checks;
-      }
-    );
+      devShells = forAllSystems (system: {
+        default = result.${system}.shell;
+      });
```

Comment:
What's the motivation for this?

Reply:
Makes sense, sounds fine to have this :)

Personally I'm using https://github.com/nix-community/nix-direnv, which also caches the shell for speed

Reply:
Sounds good to me!